src/metadata.js

```diff
@@ -1,505 +1,510 @@
+}
+
+/**
+ * Get details of a metadata object specified by its uri
+ *
+ * @method getObjectDetails
+ * @param uri uri of the metadata object for which details are to be retrieved
+ * @return {Object} object details
+ */
+export function getObjectDetails(uri) {
+    /*eslint-disable new-cap*/
+    const d = $.Deferred();
+    /*eslint-enable new-cap*/
+
+    get(uri, {
+        headers: { Accept: 'application/json' },
+        dataType: 'json',
+        contentType: 'application/json'
+    }).then(function resolveGetObject(res) {
+        d.resolve(res);
+    }, d.reject);
+
+    return d.promise();
+}
+
+/**
+ * Get identifier of a metadata object identified by its uri
+ *
+ * @method getObjectIdentifier
+ * @param uri uri of the metadata object for which the identifier is to be retrieved
+ * @return {String} object identifier
+ */
+export function getObjectIdentifier(uri) {
+    /*eslint-disable new-cap*/
+    const d = $.Deferred();
+    /*eslint-enable new-cap*/
+    function idFinder(obj) {
+        if (obj.attribute) {
+            return obj.attribute.content.displayForms[0].meta.identifier;
+        } else if (obj.dimension) {
+            return obj.dimension.content.attributes.content.displayForms[0].meta.identifier;
+        } else if (obj.metric) {
+            return obj.metric.meta.identifier;
         }
 
-        return d.promise();
-    };
-
-    /**
-     * Get uri of an metadata object, specified by its identifier and project id it belongs to
-     *
-     * @method getObjectUri
-     * @param projectId id of the project
-     * @param identifier identifier of the metadata object
-     * @return {String} uri of the metadata object
-     */
-    var getObjectUri = function(projectId, identifier) {
-        var d = $.Deferred(),
-            uriFinder = function(obj) {
-                var data = (obj.attribute) ? obj.attribute : obj.metric;
-                return data.meta.uri;
-            };
-
-        xhr.ajax('/gdc/md/'+projectId+'/identifiers', {
-            type: 'POST',
-            headers: { Accept: 'application/json' },
-            data: {
-                identifierToUri: [identifier]
-            }
-        }).then(function(data) {
-            var found = data.identifiers.filter(function(i) {
-                return i.identifier === identifier;
-            });
+        throw Error('Unknown object!');
+    }
+
+    if (!$.isPlainObject(uri)) {
+        getObjectDetails(uri).then(function resolveGetObjectDetails(data) { d.resolve(idFinder(data)); }, d.reject);
+    } else {
+        d.resolve(idFinder(uri));
+    }
+
+    return d.promise();
+}
+
+/**
+ * Get uri of an metadata object, specified by its identifier and project id it belongs to
+ *
+ * @method getObjectUri
+ * @param projectId id of the project
+ * @param identifier identifier of the metadata object
+ * @return {String} uri of the metadata object
+ */
+export function getObjectUri(projectId, identifier) {
+    /*eslint-disable new-cap*/
```

Comment:
I'd rather get rid of the new-cap rule than specify a milion of exceptions everywhere